### PR TITLE
io: reverse file content

### PR DESCRIPTION
**Pull request by ivanP91**
task two  is ready

### CODE REVIEW

File: src/com/clouway/task2/InputOutputFile.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package com.clouway.task2;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+
+public class InputOutputFile {
```
Comment (mgenov) [inline src/com/clouway/task2/InputOutputFile.java]: Doesn't sound like a good name. 

Java classes are:  InputStream, OutputStream but not InputOutputStream.

File: src/com/clouway/task2/InputOutputFile.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
+    String line;
+    try {
+      do {
+        line = in.readLine();
+        out.write(line);
+        out.newLine();
+      } while (!line.equals("."));
+    } catch (IOException io) {
+      io.printStackTrace();
```
Comment (mgenov) [inline src/com/clouway/task2/InputOutputFile.java]: The error handling is omitted.

**Comment by mgenov** [comment]
Which "task" is ready ?

File: src/com/clouway/task2/InputOutputFile.java (hunk 1 above)
Comment (ivanP91) [inline src/com/clouway/task2/InputOutputFile.java]: WriteInFile
what you think for new name on class

Comment (mgenov) [inline src/com/clouway/task2/InputOutputFile.java]: WriteInFile is a verb, not a noun.

Comment (ivanP91) [inline src/com/clouway/task2/InputOutputFile.java]: ConsoleCreateTextfile

Comment (mgenov) [inline src/com/clouway/task2/InputOutputFile.java]: Sounds even worse.

Comment (ivanP91) [inline src/com/clouway/task2/InputOutputFile.java]: TextFile

Comment (mgenov) [inline src/com/clouway/task2/InputOutputFile.java]: Name is taken, cause someone is using this name already.

Comment (ivanP91) [inline src/com/clouway/task2/InputOutputFile.java]: ConsoleFile

Comment (mgenov) [inline src/com/clouway/task2/InputOutputFile.java]: In the context of Computer Science what is the meaning of ConsoleFile?

Comment (ivanP91) [inline src/com/clouway/task2/InputOutputFile.java]: only "File" for name on class.

Comment (mgenov) [inline src/com/clouway/task2/InputOutputFile.java]: Sounds good to me.

Comment (ivanP91) [inline src/com/clouway/task2/InputOutputFile.java]: okay

File: src/com/clouway/task2/InputOutputFile.java (hunk 2 above)
Comment (ivanP91) [inline src/com/clouway/task2/InputOutputFile.java]: I removed try{} catch{}

File: src/com/clouway/task2/File.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import java.io.InputStreamReader;
+
+public class File {
+  public String file;
+
+  public File(String name) {
+    file = name;
+  }
+
+  public void writeFile() throws IOException {
```
Comment (mgenov) [inline src/com/clouway/task2/File.java]: Actually this methods first reads some content from stdin and after that the same content is written to the file till some special symbol "." is reached. 

What I'm trying to say is that name of the method doesn't say with nothing what it does.

Comment (ivanP91) [inline src/com/clouway/task2/File.java]: i must change method name ?

Comment (mgenov) [inline src/com/clouway/task2/File.java]: Yes.

Comment (ivanP91) [inline src/com/clouway/task2/File.java]: readAndWriteToStopSymbol 
new name for method

Comment (mgenov) [inline src/com/clouway/task2/File.java]: Yeah, you may do this, but the word "and" is showing that method is doing 2 things. So you can add additional helper interface about this stuff. 

Something like this, will represent input source.
```java
interface TextSource {
    boolean hasNext();
    String nextLine(); 
}
```

Whether the implementation would be like: 
```java
public class BreakingTextSource implements TextSource {
...   
   public BreakingTextSource(String breakCharacter) {

   }

   public boolean hasNext() {
         
   }
}
```

and method definition will be simple as: 
```java
void write(TextSource source) {
   while(source.hasNext()) {
         writeTextInFile(source.nextLine()); 
    }
}
```

And as you can see, method is just writing read content from text source till there is such.

Comment (ivanP91) [inline src/com/clouway/task2/File.java]: i divided method on two  "write" and "read"

Comment (ivanP91) [inline src/com/clouway/task2/File.java]: task two ready is review

File: src/com/clauway/task2/File.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package com.clauway.task2;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+
+
+public class File {
+  void write(TextSource source, String name) {
```
Comment (mgenov) [inline src/com/clauway/task2/File.java]: So, single File will write many files ?

File: src/com/clauway/task2/BreakingTextSource.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  private final String breakCharacter;
+  private String input = " ";
+
+  public BreakingTextSource(String breakCharacter) {
+
+    this.breakCharacter = breakCharacter;
+  }
+
+  public boolean hasNext() {
+    if (!input.equals(breakCharacter)) {
```
Comment (mgenov) [inline src/com/clauway/task2/BreakingTextSource.java]: I'm not sure that this will work.

Comment (mgenov) [inline src/com/clauway/task2/BreakingTextSource.java]: You are setting input to the last line that was read, so things will work. Are you writing the breaCharacter to the file or not ?

Comment (ivanP91) [inline src/com/clauway/task2/BreakingTextSource.java]: yes,writing

**Comment by mgenov** [comment]
LGTM !

**Comment by ivanP91** [comment]
next task is ready  for review

**Comment by mgenov** [comment]
Why is not in followin' PR ?

**Pull request closed without merging**